software/runtime/runtime.h

```diff
@@ -19,8 +19,12 @@ typedef uint32_t mempool_timer_t;
 
 /// Obtain the number of cores in the current cluster.
 static inline mempool_id_t mempool_get_core_count() {
+#ifdef NUM_CORES
+  return NUM_CORES;
+#else
   extern uint32_t nr_cores_address_reg;
   return nr_cores_address_reg;
```

Comment:
Why can't we read the register?

Reply:
We can. It just adds a noticeable runtime overhead, which is a bit annoying when developing software. But I don't have a strong opinion about the default here. We can drop this change if you prefer.

Reply:
Do we need this register at all (i.e., in our `crt0`)? If not, we might as well remove the register. Or we can change this function to always return `NUM_CORES` (which should always be defined).

Reply:
I believe our software already makes use of the `NUM_CORES` define somewhere else, and I don't think asking the user to recompile the code for different configurations would be asking that much. I am a bit afraid of the perceived code fragmentation with a define that is never used, but as you see fit. Up to you to decide, then you can merge this :)

Reply:
Yes, it does. We currently need it to calculate the L1 size and for static allocations. I was just reasoning that we should not remove the register nevertheless because it also has its benefits. I will change the `mempool_get_core_count` to always use the define or throw an error, so there is no confusion where the value is coming from.

That reminds me that we can also do the same optimization for the TCDM base address. I will quickly add this as well.